WordPress/gutenberg
# Site Editor: Add 'Show template' toggle when editing pages

## What?
First pass at adding a _Show template_ toggle to the site editor when editing a page. Toggling this hides the blocks belonging to the template entirely so that you're left with a minimal interface that mimics the post editor.

## Why?
Allows users to focus on their content when desired. Moreover, it is small a step towards one day unifying the post editor and site editor. See item 3 in https://github.com/WordPress/gutenberg/issues/41717.

## How?
We re-use the approach that the site editor has for editing a `wp_navigation` CPT but instead of creating a fake `core/navigation` block we create a fake `core/group` block containing a `core/post-title, `core/post-content` and `core/post-featured-image`. These blocks are shown in the same order they appear in the template and do not appear if the template doesn't have it.

## Testing Instructions

1. Go to Appearance → Editor → Pages, select or create a page, then edit it.
2. In the Page settings sidebar, click on the Template dropdown and toggle "Show template" and "Hide template"
3. Check that the page content blocks are the same between hide/show modes.
4. Hide the template and then navigate away from edit mode in the site editor by clicking on the WP logo. The template should reappear in view mode.
5. Navigate to another section, e.g., Patterns, then return to the same page you edited previously
6. The page blocks should appear as expected

Run some tests:

```
npm run test:unit packages/edit-site/src/components/block-editor/block-editor-provider/test/use-page-content-blocks.js
```

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/6458278/41792555-9c99-4b5e-86a2-24638352f361
